core/src/main/java/io/questdb/cutlass/line/http/LineHttpSender.java

```diff
@@ -0,0 +1,637 @@
+        }
+    }
+
+    private void consumeChunkedResponse(HttpClient.ResponseHeaders response) {
+        if (!response.isChunked()) {
+            return;
+        }
+        ChunkedResponse chunkedRsp = response.getChunkedResponse();
+        while ((chunkedRsp.recv()) != null) {
+            // we don't care about the response, just consume it so it won't stay in the socket receive buffer
+        }
+    }
+
+    private void escapeQuotedString(CharSequence name) {
+        for (int i = 0, n = name.length(); i < n; i++) {
+            char c = name.charAt(i);
+            switch (c) {
+                case ' ':
+                case ',':
+                case '=':
+                case '\n':
+                case '\r':
+                case '\\':
+                    request.put((byte) '\\').put((byte) c);
+                    break;
+                default:
+                    request.put(c);
+                    break;
+            }
+        }
+    }
+
+    private void escapeString(CharSequence value) {
+        for (int i = 0, n = value.length(); i < n; i++) {
+            char c = value.charAt(i);
+            switch (c) {
+                case '\n':
+                case '\r':
+                case '"':
+                case '\\':
+                    request.put((byte) '\\').put((byte) c);
+                    break;
+                default:
+                    request.put(c);
+                    break;
+            }
+        }
+    }
+
+    private void flush0(boolean closing) {
+        if (state != RequestState.EMPTY && !closing) {
+            throw new LineSenderException("Cannot flush buffer while row is in progress. Use sender.at() or sender.atNow() to finish the current row first.");
+        }
+        if (pendingRows == 0) {
+            return;
+        }
+
+        long retryingDeadlineNanos = Long.MIN_VALUE;
+        int retryBackoff = RETRY_INITIAL_BACKOFF_MS;
+        for (; ; ) {
+            try {
+                HttpClient.ResponseHeaders response = request.send(host, port);
+                response.await();
+                DirectUtf8Sequence statusCode = response.getStatusCode();
+                if (isSuccessResponse(statusCode)) {
+                    consumeChunkedResponse(response); // if any
+                    if (keepAliveDisabled(response)) {
+                        // Server has HTTP keep-alive disabled and it's closing this TCP connection.
+                        client.disconnect();
+                    }
+                    break;
+                }
+                if (isRetryableHttpStatus(statusCode)) {
+                    long nowNanos = System.nanoTime();
+                    retryingDeadlineNanos = (retryingDeadlineNanos == Long.MIN_VALUE && !closing) ? nowNanos + maxRetriesNanos : retryingDeadlineNanos;
+                    if (nowNanos >= retryingDeadlineNanos) {
+                        handleHttpErrorResponse(statusCode, response);
```

Comment:
Shouldn't we also disconnect and add a `break;` somewhere here? Currently we fall-through and go for another request spin at L346.

Reply:
`handleHttpErrorResponse()` throws. I guess naming could indicate it more explicitly.

Reply:
Oh, I forgot that. Yes, a better name wouldn't hurt.

Reply:
wdyt now?

Reply:
Looks good, thanks